docs/SIL.rst

```diff
@@ -2316,6 +2317,57 @@ The current list of interior pointer SIL instructions are:
 (*) We still need to finish adding support for project_box, but all other
 interior pointers are guarded already.
 
+Dead End Blocks
+~~~~~~~~~~~~~~~
+
+In SIL, one can express that a program is semantically expected to exit at the
+end of a block by terminating the block with an `unreachable`_. Such a block is
+called a *program terminating block* and all blocks that are post-dominated by
+blocks of the aforementioned kind are called *dead end blocks*. Intuitively, any
+path through a dead end block is known to result in program termination, so
+resources that normally would need to be released back to the system will
+instead be returned to the system upon process tear down.
+
+Since we rely on the system at these points to perform resource cleanup, we are
+able to loosen our lifetime requirements by allowing for values to not have
+their lifetimes ended along paths that end in program terminating
+blocks. Operationally, the implication for this is that end_borrow,
+destroy_value instructions are redundent, albeit legal, in dead end
+blocks. Consider the following legal SIL where we leak ``%0`` and ``%1``::
```

Comment:
> Consider the following legal SIL where we leak ``%0`` and ``%1``

The point is that this SIL does *not* leak `%0`. There's no `%1`.

Reply:
In an earlier example I had an alloc_ref and I think I removed it late while editing. I'll fix. Good catch!